### PR TITLE
Improved error messages when encountering invalid tokens

### LINKED ISSUE #13029: Improve the error message about JDL options that don't exist
Related issue: #13027. 

Brief description: when importing a JDL content with an option that doesn't exist, the current error message isn't clear enough.
It says there's a syntax error, but doesn't explicitly state the JDL content contains a bad option.
`Error: MismatchedTokenException: Expecting --> '}' <-- but found --> 'unknownOption' <--` is an example of the error message.

cc @mraible

### PR DESCRIPTION
**Pull request by MathieuAA**
Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

Fixes #13029

### CODE REVIEW

File: jdl/parsing/api.js [hunk 1]
```diff
@@ -66,13 +66,24 @@ function getCst(input, startRule = 'prog') {
...
+    }
+    const errorMessage = `${parserError.name}: ${parserError.message}`;
+    const { token } = parserError;
     const errorMessageLocation = token.tokenType !== EOF ? `\n\tat line: ${token.startLine}, column: ${token.startColumn}` : '';
     throw Error(`${errorMessage}${errorMessageLocation}`);
 }
 
+function throwErrorAboutInvalidToken(parserError) {
+    const { token } = parserError;
+    const errorMessageBeginning = `Found the invalid token '${token.image}'`;
```
Comment (mraible) [inline jdl/parsing/api.js:81]:
```suggestion
    const errorMessageBeginning = `Found an invalid token '${token.image}'`;
```

File: test/jdl/grammar/validator.spec.js [hunk 2]
```diff
@@ -876,7 +876,7 @@ describe('JDLSyntaxValidatorVisitor', () => {
                 jhiPrefix 42abc
               }
             }`)
-                        ).to.throw("Exception: Expecting --> '}' <-- but found --> 'abc'");
+                        ).to.throw("MismatchedTokenException: Found the invalid token 'abc', at line: 4 and column: 29.");
```
Comment (mraible) [inline test/jdl/grammar/validator.spec.js:879]:
```suggestion
                        ).to.throw("MismatchedTokenException: Found an invalid token 'abc', at line: 4 and column: 29.");
```

File: test/jdl/grammar/validator.spec.js [hunk 3]
```diff
@@ -917,7 +917,7 @@ describe('JDLSyntaxValidatorVisitor', () => {
                 languages [fr, en, 42]
               }
             }`)
-                        ).to.throw("MismatchedTokenException: Expecting token of type --> NAME <-- but found --> '42' <--");
+                        ).to.throw(/^MismatchedTokenException: Found the invalid token '42', at line: \d+ and column: \d+\./);
```
Comment (mraible) [inline test/jdl/grammar/validator.spec.js:920]:
```suggestion
                        ).to.throw(/^MismatchedTokenException: Found an invalid token '42', at line: \d+ and column: \d+\./);
```

File: test/jdl/grammar/validator.spec.js [hunk 4]
```diff
@@ -1389,7 +1389,7 @@ describe('JDLSyntaxValidatorVisitor', () => {
                 testFrameworks [fr, en, 42]
               }
             }`)
-                        ).to.throw("MismatchedTokenException: Expecting token of type --> NAME <-- but found --> '42' <--");
+                        ).to.throw(/^MismatchedTokenException: Found the invalid token '42', at line: \d+ and column: \d+\./);
```
Comment (mraible) [inline test/jdl/grammar/validator.spec.js:1392]:
```suggestion
                        ).to.throw(/^MismatchedTokenException: Found an invalid token '42', at line: \d+ and column: \d+\./);
```

File: test/jdl/grammar/validator.spec.js [hunk 5]
```diff
@@ -1679,7 +1679,7 @@ describe('JDLSyntaxValidatorVisitor', () => {
                 deployment {
                   ${type} [fr, en, @123]
               }`)
-                            ).to.throw("MismatchedTokenException: Expecting token of type --> NAME <-- but found --> '@' <--");
+                            ).to.throw(/^MismatchedTokenException: Found the invalid token '@', at line: \d+ and column: \d+\./);
```
Comment (mraible) [inline test/jdl/grammar/validator.spec.js:1682]:
```suggestion
                            ).to.throw(/^MismatchedTokenException: Found an invalid token '@', at line: \d+ and column: \d+\./);
```

File: test/jdl/jdl-importer.spec.js [hunk 6]
```diff
@@ -1977,5 +1977,34 @@ use mapstruct, elasticsearch for A, B except C`;
...
+                    const importer = createImporterFromContent(content);
+                    importer.import();
+                } catch (error) {
+                    caughtError = error;
+                }
+            });
+
+            it('should report it', () => {
+                expect(caughtError.message).to.equal(
+                    "MismatchedTokenException: Found the invalid token 'unknownOption', at line: 5 and column: 5.\n\tPlease make sure your JDL content does not use invalid characters, keywords or options."
```
Comment (mraible) [inline test/jdl/jdl-importer.spec.js:2005]:
```suggestion
                    "MismatchedTokenException: Found an invalid token 'unknownOption', at line: 5 and column: 5.\n\tPlease make sure your JDL content does not use invalid characters, keywords or options."
```

**Review by mraible — CHANGES_REQUESTED** [review]

File: jdl/parsing/api.js (hunk 1 above)
Comment (MathieuAA) [reply jdl/parsing/api.js]: done

File: test/jdl/grammar/validator.spec.js (hunk 2 above)
Comment (MathieuAA) [reply test/jdl/grammar/validator.spec.js]: done

File: test/jdl/grammar/validator.spec.js (hunk 3 above)
Comment (MathieuAA) [reply test/jdl/grammar/validator.spec.js]: done

File: test/jdl/grammar/validator.spec.js (hunk 4 above)
Comment (MathieuAA) [reply test/jdl/grammar/validator.spec.js]: done

File: test/jdl/grammar/validator.spec.js (hunk 5 above)
Comment (MathieuAA) [reply test/jdl/grammar/validator.spec.js]: done

File: test/jdl/jdl-importer.spec.js (hunk 6 above)
Comment (MathieuAA) [reply test/jdl/jdl-importer.spec.js]: done

**Review by mraible — APPROVED** [review]

**Comment by MathieuAA** [comment]
Thanks for reviewing everything :)

**Pull request merged**